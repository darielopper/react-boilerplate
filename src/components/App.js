import React from 'react'

const App = () => (
    <div>
        <h1>Hello World, from React App!!</h1>
    </div>
)

export default App