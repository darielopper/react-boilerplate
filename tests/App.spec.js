import React from 'react'
import App from '../src/components/App'
import Adapter from 'enzyme-adapter-react-16'
import sinon from 'sinon' //For mockery and click spies
import { shallow, configure, mount, render } from 'enzyme'

configure({ adapter: new Adapter() })

describe('App component', () => {
    test('should shallow correctly', () => {
        expect(shallow(
            <App />
        )).toMatchSnapshot()
    })

    test('should mount correctly', () => {
        expect(mount(
            <App />
        )).toMatchSnapshot()
    })

    test('should render correctly', () => {
        expect(render(
            <App />
        )).toMatchSnapshot()
    })

    test('have default text', () => {
        const wrapper = shallow(<App />)
        expect(wrapper.find('h1').text()).toContain('Hello World')
    })
})