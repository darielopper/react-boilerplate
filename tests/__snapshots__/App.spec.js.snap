// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should mount correctly 1`] = `ReactWrapper {}`;

exports[`App component should render correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "data": "Hello World, from React App!!",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "h1",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`App component should shallow correctly 1`] = `ShallowWrapper {}`;
